openxlsx.Rmd

```diff
@@ -0,0 +1,581 @@
+
+While dashboards and infographics are a great way of showing results and delivering messages they don't suit everyone. Many of our customers can't access dashboards due to IT setup in their organisations and others may just want the data in a spreadsheet to take away and do their own analysis. 
+
+```{r openxlsx-horst, echo=FALSE, fig.align='centre', out.width="100%"}
+knitr::include_graphics("images/openxlsx/openxlsx_horst.jfif")
+```
+
+
+## Creating a workbook
+We will start by creating a blank workbook and then slowly build it up by adding tables, charts styles and formatting. 
+
+Let's start by creating a blank excel template that you build on. Note that you can't view it as you go along, only once it is saved out, so you have to keep track of what you are adding. 
+
+````
+wb <- createWorkbook()
+````
+
+### Adding worksheets
+Before we can add any data to the workbook we have to create worksheets for the data to go in. Note that the order you add the sheets to the workbook is the order they appear in, this is not set in code. We will add 3 worksheets:
+
+* A notes page
+* A data page
+* A charts page
+
+Notes page:
+````
+# Add Notes Page
+addWorksheet(
+# Name of the workbook we want to add the sheet to
+wb = wb,
+
+# Define the name of the worksheet 
+sheetName = "Notes",
+
+# Do we want to show gridlines? The default is TRUE so we have to specify if we don't want them
+gridLines = FALSE,
+
+# Add headers - as a character vector for positions left, centre and right
+# You can use some in built things
+# (such as page numbers, date, time, filename etc)
+header = c("&[Date]", NA, NA))
+````
+
+Data Page:
+````
+# Add Data Page
+addWorksheet(
+# Name of the workbook we want to add the sheet to
+wb = wb,
+
+# Define the name of the worksheet 
+sheetName = "Data",
+
+# Do we want to show gridlines? The default is TRUE so we have to specify if we don't want them
+gridLines = FALSE,
+
+# Colour of the tab - either use from colours() list or use hex code (optional)
+tabColour = "red", # Uses in built colour
+
+# Add headers - as a character vector for positions left, centre and right
+# You can use some in built things
+# (such as page numbers, date, time, filename etc)
+header = c("ODD HEAD LEFT", "ODD HEAD CENTER", "ODD HEAD RIGHT"))
+````
+
+Charts Page:
+````
+# Add Charts Page
+addWorksheet(
+# Name of the workbook we want to add the sheet to
+wb = wb,
+
+# Define the name of the worksheet 
+sheetName = "Charts",
+
+# Do we want to show gridlines? The default is TRUE so we have to specify if we don't want them
+gridLines = FALSE,
+
+# Colour of the tab - either use from colours() list or use hex code (optional)
+tabColour = "blue", # Uses in built colour
+
+# Add headers - as a character vector for positions left, centre and right
+# You can use some in built things
+# (such as page numbers, date, time, filename etc)
+header = c("ODD HEAD LEFT", "ODD HEAD CENTER", "ODD HEAD RIGHT"))
+````
+
```

Comment:
I realise you've got comments covering what's happening with each of the arguments. However, you could include a section before using the `createWorkbook()` and `addWorksheet()` functions explaining what they do and what the arguments are. This makes it more of a course/good for reference, rather than tutorial alone. Then the same for the other functions used later in the course.

Reply:
Hi @rmccreath I've added this for one or two of the functions so let me know what you think before I keep going!